src/bin/miri.rs

```diff
@@ -228,7 +231,37 @@ fn main() {
                     miri_config.check_alignment = miri::AlignmentCheck::Symbolic;
                 }
                 "-Zmiri-disable-isolation" => {
-                    miri_config.communicate = true;
+                    if let Some(true) = isolation_enabled {
+                        panic!(
+                            "-Zmiri-disable-isolation cannot be used along with -Zmiri-isolation-error"
+                        );
+                    } else {
+                        isolation_enabled = Some(false);
+                    }
+                    miri_config.isolated_op = miri::IsolatedOp::Allow;
+                }
+                arg if arg.starts_with("-Zmiri-isolation-error=") => {
+                    if let Some(false) = isolation_enabled {
+                        panic!(
+                            "-Zmiri-isolation-error cannot be used along with -Zmiri-disable-isolation"
+                        );
+                    } else {
+                        isolation_enabled = Some(true);
+                    }
+
+                    miri_config.isolated_op = match arg
+                        .strip_prefix("-Zmiri-isolation-error=")
+                        .unwrap()
+                    {
+                        "abort" => miri::IsolatedOp::Reject(miri::RejectOpWith::Abort),
+                        "ignore" => miri::IsolatedOp::Reject(miri::RejectOpWith::NoWarning),
```

Comment:
I think we should rename "ignore", but I am not sure what to rename it to... maybe "hide"? I want it to be clear that isolation is still enforced.

Reply:
Listing out options I could think of:
- `-Zmiri-isolation-error=hide`
- `-Zmiri-isolation-error=no-warn`
- `-Zmiri-isolation-error=silent`

Reply:
One more
`-Zmiri-isolation-error=Shh..I_know_what_I_am_doing` 😄

Reply:
From the proposals so far, I think I prefer "hide", so shall we go with that?